enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,611 @@
+   feedback.
+
+## Design Details
+
+### Proposed API changes
+
+`ScyllaDBMonitoring` CRD will be extended with the following changes:
+
+1. Add a new `spec.components.prometheus.mode` field to allow users to choose the mode of the Prometheus component.
+   The field will be an enum, defaulting to `OperatorManaged` (i.e., Operator-managed Prometheus is enabled by default).
+   When set to `UserManaged`, the Operator will not deploy a managed Prometheus instance. Only the scraping and alerting
+   resources will be created (`ServiceMonitor`s and `PrometheusRule`s) to allow the external Prometheus instance to
+   scrape metrics and alerts from the ScyllaDB cluster.
+
+    ```go
+    // PrometheusMode describes the mode of the Prometheus instance.
+    // +kubebuilder:validation:Enum="Managed";"External"
+    type PrometheusMode string
+
+    const (
+        // PrometheusModeManaged defines a mode where a `Prometheus` object is created as a child of a `ScyllaDBMonitoring`
+        // object. `ServiceMonitor` and `PrometheusRule` resources are also created to configure scraping and alerting.
+		// This mode requires a Prometheus Operator to be installed in the cluster.
+        PrometheusModeManaged PrometheusMode = "Managed"
+   
+        // PrometheusModeExternal defines a mode where no `Prometheus` child object is created, but `ServiceMonitor` and
+        // `PrometheusRule` objects are still created to configure scraping and alerting.
+        // This mode requires a Prometheus Operator to be installed in the cluster, along with a `Prometheus` instance
+        // configured to reconcile `ServiceMonitor` and `PrometheusRule` resources.
+        PrometheusModeExternal PrometheusMode = "External"
+    )
+    
+    type PrometheusSpec struct {
+        // mode defines the mode of the Prometheus instance. It's an immutable field.
+        // +kubebuilder:default:="OperatorManaged"
+        // +optional
+        Mode PrometheusMode `json:"mode,omitempty"`
+        ...
+    }
+    ```
+
+   Externally managed Prometheus instances can be configured to reconcile `ServiceMonitor` and `PrometheusRule`
+   resources by setting `ruleSelector` and `serviceMonitorSelector` fields in
+   the [Prometheus](https://prometheus-operator.dev/docs/api-reference/api/#monitoring.coreos.com/v1.Prometheus) spec.
+   We will document that these fields have to match the
+   `scylla-operator.scylladb.com/scylladbmonitoring-name: "{{ .scyllaDBMonitoringName }}"`
+   label in order to work, as that's the label the Operator adds to the `ServiceMonitor` and `PrometheusRule` resources
+   it creates.
+
+   The added `mode` field will be immutable to prevent users from switching between managed and external Prometheus
+   modes.
+   Implementation of switching modes would require additional work to ensure the managed resources are
+   properly cleaned up, and it's not worth the effort at this point.
```

Comment:
Operator is responsible for cleaning resources and every our controller does that. It's simple because they all match particular label selector. 
I do think this field should be immutable, but definitely not for cleaning burden reason.

Reply:
Reworded.